### PR TITLE
Add SessionHop to allow infinite levels of role assumption goodness

### PR DESCRIPTION
**Pull request by theherk**

### CODE REVIEW

**Comment by theherk** [comment]
@defond0, @mochamanbj, @yousefcodes

File: aidews.go [hunk 1]
```diff
@@ -5,31 +5,44 @@ package aidews
...
+// then get a session in that region using the credentials from the STS call.
+//
+// All Sessions are constructed using the SharedConfigEnable setting allowing
+// the use of local credential resolution.
+func Session(region, roleARN *string) *session.Session {
+	return session(nil, region, roleARN)
+}
+
+// SessionHop returns an aws session constructed from a given Session.
+// This is very similar to Session, but allows hopping from a given session, to
```
Comment (defond0) [inline aidews.go]: It might be good to use the assumerole terminology here for the sake of clarity

**Review by defond0 — CHANGES_REQUESTED** [review]
Comment update

File: aidews.go (hunk 1 above)
Comment (theherk) [reply aidews.go]: Very good, sir. It shall be done.

Comment (theherk) [reply aidews.go]: Updated the documentation, and added the following example text.

``` go
start := Session(region, startingRoleARN)
hop1 := SessionHop(start, region, hop1ARN)
hop2 := SessionHop(hop1, region, hop2ARN)
destination := SessionHop(hop2, region, destARN)
```

**Pull request merged**